src/main/java/org/openstreetmap/atlas/generator/tools/spark/SparkJob.java

```diff
@@ -79,19 +78,31 @@
 
     private transient JavaSparkContext context;
 
-    public static Resource resource(final String path, final Map<String, String> configurationMap)
+    /**
+     * Get resources from a path
+     *
+     * @param path
+     *            The resource path
+     * @param configurationMap
+     *            The configuration for {@link FileSystemCreator#get(String, Configuration)}
+     * @return A {@link ResourceCloseable} (this should be closed on finish)
+     */
+    public static ResourceCloseable resource(final String path,
+            final Map<String, String> configurationMap)
     {
         if (path.startsWith("http"))
         {
             return new GetResource(path);
         }
-        try (FileSystem fileSystem = new FileSystemCreator().get(path, configurationMap))
+        // This is closed by the ResourceCloseable class
+        final FileSystem fileSystem = new FileSystemCreator().get(path, configurationMap);
```

Comment:
With that change, we would never close FileSystems anymore, just resources?

Reply:
This is a fair point. There are a few paths where the path doesn't exist, or there is an exception thrown, where the filesystem isn't closed. I'll fix this.